### PR TITLE
add comment-item component documentation

### PR DESCRIPTION
**Pull request by WenInCode**
# What's in this PR?
- adds documentation for the `comment-item` component
- minor refactors

## Reference
Issue: #223

### CODE REVIEW

File: app/components/comment-item.js [hunk 1]
```diff
@@ -1,54 +1,155 @@
...
+  currentUserIsCommentAuthor: computed('currentUserId', 'commentAuthorId', function() {
+    let userId = parseInt(get(this, 'currentUserId'), 10);
+    let authorId = parseInt(get(this, 'commentAuthorId'), 10);
+
+    return isEqual(userId, authorId);
   }),
 
+  /**
+    Overrides the init method, sets the `isEditing` property to `false`,
+    prefetches menthions and calls the `super` on `init`.
```
Comment (npendery) [inline app/components/comment-item.js]: Spelling fix for `mentions`

File: app/components/comment-item.js [hunk 2]
```diff
@@ -1,54 +1,155 @@
...
     },
 
+    /**
+      Action that sets the corresponding comment to edit mode.
+
+      @method edit
+     */
     edit() {
-      this.set('isEditing', true);
+      set(this, 'isEditing', true);
```
Comment (npendery) [inline app/components/comment-item.js]: This doesnt really have to do with the docs, but couldnt we condense the cancel and edit actions to be just one toggle action?
```
toggleEdit() {
  this.toggleProperty('isEditing');
}
```

**Comment by npendery** [comment]
Small comments but looks good! 🚢 🇮🇹

File: app/components/comment-item.js (hunk 2 above)
Comment (WenInCode) [inline app/components/comment-item.js]: Good idea - thanks!

**Pull request merged**